functorch/_src/eager_transforms.py

```diff
@@ -243,12 +243,15 @@ def vjp(func: Callable, *primals, has_aux=False):
             primals_out = func(*diff_primals)
 
             if has_aux:
+                if not (isinstance(primals_out, tuple) and len(primals_out) == 2):
+                    raise TypeError("Function output should be a tuple: (output, aux) if has_aux is True")
```

Comment:
Also: nit: Should we say "vjp(f, *primals): output of function `f` should be a ..." to make it clearer where the error is coming from? If you agree then we should also update the test to check for this.